src/feature/gui/gui-runner.c

```diff
@@ -656,7 +657,12 @@ void mGUIRun(struct mGUIRunner* runner, const char* path) {
 				runner->core->reset(runner->core);
 				break;
 			case RUNNER_SAVE_STATE:
-				mCoreSaveState(runner->core, item->data.v.u >> 16, SAVESTATE_SCREENSHOT | SAVESTATE_SAVEDATA | SAVESTATE_RTC | SAVESTATE_METADATA);
+				struct mGUIBackground* gbaBackground = (struct mGUIBackground*) stateSaveMenu.background;
```

Comment:
Don't define a variable in a case statement without declaring a block (putting the whole thing in `{}`)--I generally just avoid declaring variables if I can avoid it, or hoist the declaration. It'll get optimized anyway.

Reply:
Yeah, [I figured it would get optimized](https://github.com/mgba-emu/mgba/pull/2929/commits/5e3e71df3e5ff03ed63d7bf6234451a9c7ee48c1) anyways! I'll just remove it! If I consider your idea in [this comment](https://github.com/mgba-emu/mgba/pull/2929#discussion_r1199835438), then I think we can just remove the need for the variable entirely.

Reply:
I saw that, I was just agreeing haha. That type of optimization is specifically called common subexpression elimination, and is pretty basic.

Reply:
Did not know about its official name, TIL! Thanks for the link!